internal/services/servicenetworking/validate/application_load_balancer_subnet_association_name.go

```diff
@@ -0,0 +1,14 @@
+package validate
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+func ApplicationLoadBalancerSubnetAssociationName() pluginsdk.SchemaValidateFunc {
+	return validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9_.-]{0,62}[a-zA-Z0-9]$`),
```

Comment:
This regex is slightly incorrect for the description below, it won't allow a name of 1 character. Can we split this out into separate checks for the limitations to make it easier to maintain and report the correct violation to the users?